TypeSystem/flow-analysis/promotion_via_assignment_A06_t01.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a variable `x` is promotable via assignment of an
+/// expression of type `T` given variable model `VM` if
+/// - `VM = VariableModel(declared, promoted, tested, assigned, unassigned,
+///     captured)`
+/// - and captured is false
+/// - and `S` is the current type of `x` in `VM`
+/// - and `T <: S` and not `S <: T`
+/// - and `T` is a type of interest for `x` in `tested`
+///
+/// @description Checks static type of a variable after assignment with a
+/// supertype of the declared type as type of interest.
+/// @author sgrekhov22@gmail.com
+/// @issue 60620, 60622
+
+import '../../Utils/static_type_helper.dart';
+
+main() {
+  String s = "a";
+  if (s is Pattern) {}
+  s = "";
```

Comment:
Same difficulty again.

Reply:
And this is a test for sdk#60620, sdk#60622. May be to change the description to "Checks that if a variable `x` of type `T` such that `T <: S` has `S` as a type of interest, then assignment type `T` to `x` doesn't demote its type  `S`"?

Reply:
With the desired spec updates, this will promote `s` to `Pattern`. Let's just expect that.

Reply:
The problem with `test2` is that it is not a test which is based on language related expectations, it's specifically a #60620 regression test. From a language perspective there is absolutely no reason to expect that `s = ""` should equip `s` with a new type: It already has type `String` based on the declaration (and no promotions), and the assigned expression has static type `String`.

So it's kind of crazy to expect that it would get another type (such as `Pattern`) based on this situation.

It makes sense in a regression test (because we've actually seen the crazy behavior in real life!), but it does not really make sense as a language test.

Anyway, let's keep it just because this PR has been lingering forever. ;-)